src/js/common/brush.js

```diff
@@ -0,0 +1,95 @@
+function mg_create_brushing_pattern(args, range) {
+  var extentRect;
```

Comment:
Please use `const` and the ternary operator here:

```js
const extentRect = (d3.select(args.target).select('.mg-extent').size()) ? 
    d3.select(args.target).select('.mg-extent') : 
    d3.select(args.target)
      .select('.mg-rollover-rect, .mg-voronoi')
      .insert('g', '*')
      .classed('mg-brush', true)
      .append('rect')
      .classed('mg-extent', true);
```

In general we should try to use `let` or `const` for new code (http://airbnb.io/javascript/#references--prefer-const)